xcat3/plugins/control/openbmc.py

```diff
@@ -0,0 +1,94 @@
+from oslo_log import log
+from xcat3.plugins.control import base
+from xcat3.common import exception
+from xcat3.common import states
+from xcat3.common import rest
+
+import json
+import pdb
+
+LOG = log.getLogger(__name__)
+
+url_dict  = {"set_power_state" : "/xyz/openbmc_project/state/host0/attr/RequestedHostTransition",
+             "get_power_state"  : "/xyz/openbmc_project/state/host0" }
+
+data_dict = {states.POWER_COMMAND_ON   : "xyz.openbmc_project.State.Host.Transition.On",
+             states.POWER_COMMAND_OFF  : "xyz.openbmc_project.State.Host.Transition.Off",
+             states.POWER_COMMAND_RESET: "xyz.openbmc_project.State.Host.Transition.Reboot"}
+
+class OPENBMCPlugin(base.ControlInterface):
+
+    def validate(self, node):
+        bmc = node.control_info.get('bmc')
+        bmcusername = node.control_info.get('bmcusername')
+        bmcpassword = node.control_info.get('bmcpassword')
+        if not bmc:
+            raise exception.MissingParameterValue(
+                _("OPENBMC address was not specified."))
+        if not bmcusername:
+            raise exception.MissingParameterValue(
+                _("OPENBMC username was not specified."))
+        if not bmcpassword:
+            raise exception.MissingParameterValue(
+                _("OPENBMC password was not specified."))
+        return
+
+    def login(self, node):
+        login_url = 'https://' + node.control_info['bmc'] + '/login'
+        login_data = { "data": [ node.control_info['bmcusername'], node.control_info['bmcpassword'] ] }
+        session = rest.login(login_url, login_data)   
+
+        return session
+
+    def power_state_request(self, node, session):
```

Comment:
Actually, there is no relationship between login and get/set power state, that only depends on how you call the login method and send the request to the target url. You should not do it like this.
Can you transfer the session as a argument of the method?  Or consider the session as a member of the class or object?

One more comment, does the session have timeout design, will it happen if `get_power_state` get the result, but session timeout ( just assume get_power_status takes a long time), then you use the same session to `set_power_state`?  This is only some thought, I just wonder if it will be the timeout error at openbmc side so that and we need some auto retry mechanism to solve this.